### PR TITLE
remove variable zero cones before performing algorithm

### PR DESCRIPTION
**Pull request by chriscoey**
so I haven't actually tested this on a problem with zero cones, but @mlubin can you run it with some of the test instances?

### CODE REVIEW

**Comment by chriscoey** [comment]
actually hopefully the SDP tests will have variable zero cones. now I'm making it run sdp tests

**Comment by mlubin** [comment]
This should have a unit test before we merge.

**Comment by chriscoey** [comment]
yeah I know. is there a way to make convex.jl put a zero variable in? or should we just load a MPB conic format problem directly?

**Comment by mlubin** [comment]
I don't think we can get convex.jl to generate one of these. I would just cook up a tiny 3-variable problem with a zero cone.

**Comment by mlubin** [comment]
max x_1
s.t. (x_3,x_2,x_1) in SOC (as constriant cone)
       x_2 in Zero
       x_3 <= 5

File: test/conictest.jl [hunk 1]
```diff
@@ -76,13 +76,39 @@ function runconictests(mip_solver_drives, mip_solver, conic_solver)
...
 
            @fact problem.optval --> roughly(8.0, TOL)
            @fact problem.status --> :Optimal
        end
     end
 
+    facts("Variable in zero cone problem") do
+        context("With $algorithm, $(typeof(mip_solver)) and $(typeof(conic_solver))") do
+            m = MathProgBase.ConicModel(PajaritoSolver(mip_solver_drives=mip_solver_drives, mip_solver=mip_solver, cont_solver=conic_solver, log_level=0))
+            MathProgBase.loadproblem!(m,
```
Comment (mlubin) [inline test/conictest.jl]: Could you comment the math version of this problem so it's easier to read?

**Comment by chriscoey** [comment]
ok there were numerous little bugs before. hopefully the code is now in a state that it should run on the large complex instances. @mlubin if you can run those tonight or tomorrow, even just a few, then I can fix more bugs Sunday

**Comment by chriscoey** [comment]
I still get a seg fault on the BC SDP test with Mosek, strangely when creating or loading a conic model with mosek solver. Happens whether I'm on master of all relevant packages or not.

**Comment by chriscoey** [comment]
I'm going to merge this when tests pass because last merged PR had a couple bugs

**Comment by chriscoey** [comment]
looks like the rotated SOC problem failed with NLP algorithm and ipopt (OA and BC), and no other fails. @mlubin, @emreyamangil 

also, I need to remove all the debugging printing

**Comment by mlubin** [comment]
Are there tests now that would've caught the bugs you fixed?

**Comment by chriscoey** [comment]
Yeah the tests that were under sdp but are now moved to conic

**Comment by mlubin** [comment]
You can skip specific conic tests with nlp solvers if they're failing.

**Pull request merged**